File: 64. Minimum Path Sum/dp.py
class Solution:
    def minPathSum(self, grid: List[List[int]]) -> int:
        H, W = len(grid), len(grid[0])
        for y in range(H):
            for x in range(W):
                if y == 0 and x == 0:
                    continue;
                else:
                    cost = grid[y][x]
                    left = (cost + grid[y-1][x]) if y > 0 else sys.maxsize
                    right = (cost + grid[y][x-1]) if x > 0 else sys.maxsize
                    
                    grid[y][x] = min(left, right)
        
        return grid[H-1][W-1];
        
        

File: 64. Minimum Path Sum/dp.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
 var minPathSum = function(grid) {
  
    const H = grid.length;
    const W = grid[0].length;
    
    for(let y = 0; y < H; y++){
        for(let x = 0; x < W; x++){
            if(y === 0 && x === 0) continue;
            const cost = grid[y][x]
            const top = (y > 0) ? grid[y-1][x] + cost : Infinity;
            const left = (x > 0) ? grid[y][x-1] + cost : Infinity;

            grid[y][x] = Math.min(top, left)
        }
    }
    
    return grid[H-1][W-1];
};